src/users/users.service.ts

```diff
@@ -3,32 +3,49 @@ import { InjectModel } from '@nestjs/mongoose';
 import { JwtService } from '@nestjs/jwt';
 import { Model } from 'mongoose';
 import * as bcrypt from 'bcrypt';
+import Stripe from 'stripe';
 
 import { CreateUserDto } from './dto/create-user.dto';
 import { Users } from './schemas/users.schema';
 import { ConfigService } from '@nestjs/config';
 import { InjectStripe } from 'nestjs-stripe';
-import Stripe from 'stripe';
+import { RestaurantService } from '../restaurant/restaurant.service';
+import { Restaurant } from '../restaurant/schemas/restaurant.schema';
 
 @Injectable()
 export class UsersService {
   constructor(
     @InjectStripe() private readonly stripeClient: Stripe,
     @InjectModel(Users.name) private usersModel: Model<Users>,
+    @InjectModel(Restaurant.name)
+    private readonly restaurantModel: Model<Restaurant>,
     private readonly jwtService: JwtService,
     private readonly configService: ConfigService,
+    private readonly restaurantService: RestaurantService,
   ) {}
 
-  async signUp(dto: CreateUserDto): Promise<Users> {
-    const salt = await bcrypt.genSalt();
-    const hash = await bcrypt.hash(dto.password, salt);
+  async creatAccount(dto: {
+    telegramId?: string;
+    name?: string;
+    password?: string;
+    email?: string;
+    restaurantId?: string;
+  }): Promise<Users> {
+    if (dto.password) {
+      const salt = await bcrypt.genSalt();
+      const hash = await bcrypt.hash(dto.password, salt);
 
-    const newUser = new this.usersModel({
-      email: dto.email,
-      password: hash,
-    });
+      return new this.usersModel({
```

Comment:
assign property to the user object and at the end of method call the create method with user obj